ui-src/src/store.tsx

```diff
@@ -94,6 +105,13 @@ export function reducer (oldState: StoreState = defaultState, action: ReduxActio
     case 'FETCH_CURRENT_GAMES': {
       return {...state, allGames: Map(action.games) }
     }
+    case 'UPDATE_IDENTITIES': {
+      // console.log("Updating IDs: ", Map([state.identities, action.identities])
+      const newIdentities = Map(action.identities);
+      const oldIdentities = Map(state.identities);
+
+      return{...state, identities: newIdentities.merge(oldIdentities) as Map<Hash, string> }
```

Comment:
Might be faster to do it the other way around, merge newIdentities into oldIdentities